### ISSUE TITLE
Connect allows uploading RSA root cert, then fails validation when trying to upload new cert

### ISSUE THREAD

**Issue by rickducott** [hashicorp/consul#4965]
#### Overview of the Issue

Using the consul connect golang API, I can set the root cert config using an RSA root and private key, but then I cannot change the root certificate again. When trying to upload a new EC cert, I get validation errors related to RSA. 

#### Reproduction Steps

1. Create an RSA cert and private key
`openssl req -new -x509 -nodes -newkey rsa:2048 -keyout %s -out %s -days 365 -subj /C=US/ST=Massachusetts/L=Cambridge/O=Org/CN=www.example.com`

2. Upload to connect using the Go API:
```
innerConfig := make(map[string]interface{})
innerConfig["LeafCertTTL"] = "1h"
innerConfig["PrivateKey"] = string(caKeyBytes)
innerConfig["RootCert"] = string(caCertBytes)
innerConfig["RotationPeriod"] = "2160h"
conf := &api.CAConfig{
	Provider: "consul",
	Config:   innerConfig,
}
client.Connect().CASetConfig(conf, &writeOpts)
```
This succeeds. 

3. Now try to upload a new cert, either EC or RSA, fails with:
`panic: Unexpected response code: 500 (invalid key type: *rsa.PublicKey)`

Checking the code, it seems like RSA certs are not intended to be supported, but the validation doesn't happen until after the (invalid) RSA certificate has been accepted. 

### Consul info for both Client and Server

Using consul Go API, repro'ed on Consul 1.4.0. 

### Operating system and Environment details

Minishift with Kubernetes 1.12

### Log Fragments

When uploading the RSA cert:
``` 
2018/11/15 21:33:05 [INFO] connect: CA rotated to new root under provider "consul"
```

When uploading a new cert, RSA or EC: 
```
2018/11/15 21:33:22 [ERR] http: Request PUT /v1/connect/ca/configuration, error: invalid key type: *rsa.PublicKey from=127.0.0.1:43514
```

**Comment by banks** [comment]
Thanks @rickducott that's a bug indeed. Internally the CA only generates EC certs for services currently so most code only deals with EC signers etc. But we missed something here when adding partial support for RSA CA certs. I suspect the config path supports RSA but something in the rotation path doesn't so testing with an RSA cert initially works fine but not if you then try to rotate.

CC @kyhavlov

**Comment by tradel** [comment]
Hi @rickducott. I recently fixed this as a byproduct of getting ready to support other CA providers in Consul. PR #6055 should be part of the 1.6.0-rc1 release coming out this week. I hope you'll be able to retest with an RSA key and confirm that it works now.

**Comment by pearkes** [comment]
Per https://github.com/hashicorp/consul/pull/6055 this should be fixed and will be in the upcoming 1.6.0 release, if you see it again in that or a pre-release please let us know.

**Comment by rickducott** [comment]
Thanks!
